### DISCUSSION TITLE
github图床。变量有问题，github不允许设置github开头的密钥

### DISCUSSION THREAD

**Discussion by xinkeji** [LetTTGACO/elog#58] [General]
github图床。变量有问题，github不允许设置github开头的密钥

**Comment by LetTTGACO** [comment]
Github_TOKEN 变量不需要设置到环境变量，流水线会自动获取到

**Reply by xinkeji** [reply]
已经设置了，但是依旧用不了

**Reply by LetTTGACO** [reply]
可以提一个 issue 将相关elog配置文件和流水线文件贴一下

**Reply by xinkeji** [reply]
![Screenshot_2024-01-16-16-18-10-811_com.miui.gallery.jpg](https://github.com/LetTTGACO/elog/assets/33138729/e62e6ce8-b97b-436d-922c-61ac465b2b87)

**Reply by LetTTGACO** [reply]
github action 文件我看看

**Reply by LetTTGACO** [reply]
还有具体的日志报错，建议用电脑提一个 issue bug，这样会更快解决问题

**Comment by xinkeji** [comment]
具体怎么搞啊，我感觉是我自己设置有问题，我就设置了elog.config.js和elog.env



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年1月16日(周二) 下午4:39
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [LetTTGACO/elog] github图床。变量有问题，github不允许设置github开头的密钥 (Discussion #58)




 
还有具体的日志报错，建议用电脑提一个 issue bug，这样会更快解决问题

**Comment by xinkeji** [comment]
主要是没有报错，就是不喜欢文件



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年1月16日(周二) 下午4:39
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [LetTTGACO/elog] github图床。变量有问题，github不允许设置github开头的密钥 (Discussion #58)




 
还有具体的日志报错，建议用电脑提一个 issue bug，这样会更快解决问题

**Reply by LetTTGACO** [reply]
请在 [bug issues](https://github.com/LetTTGACO/elog/issues/new?assignees=&labels=bug&projects=&template=bug_report.yml) 中粘贴 elog.config.js 和.github/workflows下的 github actions 的文件内容

没有具体信息 我也无法帮你定位问题

**Comment by xinkeji** [comment]
[![Deploy Elog-Hexo Public To Pages](https://github.com/xinkeji/hexo/actions/workflows/Deploy%20Elog-Hexo%20Public%20To%20Pages.yml/badge.svg)](https://github.com/xinkeji/hexo/actions/workflows/Deploy%20Elog-Hexo%20Public%20To%20Pages.yml)



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年1月16日(周二) 下午4:55
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [LetTTGACO/elog] github图床。变量有问题，github不允许设置github开头的密钥 (Discussion #58)




 
请在 bug issues 中粘贴 elog.config.js 和.github/workflows下的 github actions 的文件内容
 
没有具体信息 我也无法帮你定位问题

**Reply by LetTTGACO** [reply]
github不允许设置github开头的密钥，那么就改名，只要一一对应就行，你可以使用以下代码进行替代
```
image: {
    enabled: true,
    platform: 'github',
    github: {
      user: process.env.ELOG_GITHUB_USER,
      token: process.env.GITHUB_TOKEN,
      repo: process.env.ELOG_GITHUB_REPO,
      prefixKey: '',
      host: 'cdn.jsdelivr.net'
    }
  },
```

然后在Github 流水线中注入`ELOG_GITHUB_USER`和`ELOG_GITHUB_REPO`，`GITHUB_TOKEN`不需要配置，流水线可自动获取
![image](https://github.com/LetTTGACO/elog/assets/37357188/1b7595e3-eeff-4f72-9009-cb09fac316b8)

**Reply by LetTTGACO** [reply]
我都在这里贴出来了 所有用到的环境变量 都需要在这张图中预先定义（除了GITHUB_TOKEN）

**Comment by xinkeji** [comment]
[![Deploy Elog-Hexo Public To Pages](https://github.com/xinkeji/hexo/actions/workflows/Deploy%20Elog-Hexo%20Public%20To%20Pages.yml/badge.svg)](https://github.com/xinkeji/hexo/actions/workflows/Deploy%20Elog-Hexo%20Public%20To%20Pages.yml)




---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年1月16日(周二) 下午5:25
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [LetTTGACO/elog] github图床。变量有问题，github不允许设置github开头的密钥 (Discussion #58)




 
github不允许设置github开头的密钥，那么就改名，只要一一对应就行，你可以使用以下代码进行替代
 image: {     enabled: true,     platform: 'github',     github: {       user: process.env.ELOG_GITHUB_USER,       token: process.env.GITHUB_TOKEN,       repo: process.env.ELOG_GITHUB_REPO,       prefixKey: '',       host: 'cdn.jsdelivr.net'     }   },  
然后在Github 流水线中注入ELOG_GITHUB_USER和ELOG_GITHUB_REPO，不需要配置，流水线可自动获取
 image.png (view on web)

**Reply by LetTTGACO** [reply]
不是 你还是没在流水线注入啊，改了配置文件，流水线里面没配置有什么用

你还是去看一遍 GITHUB ACTION 教程吧，你这样一步步问问题也浪费大家的时间
https://docs.github.com/zh/actions/quickstart
https://juejin.cn/post/6960126908180725773

**Comment by xinkeji** [comment]
prefixKey这个咋设置啊



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年1月16日(周二) 下午5:25
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [LetTTGACO/elog] github图床。变量有问题，github不允许设置github开头的密钥 (Discussion #58)




 
github不允许设置github开头的密钥，那么就改名，只要一一对应就行，你可以使用以下代码进行替代
 image: {     enabled: true,     platform: 'github',     github: {       user: process.env.ELOG_GITHUB_USER,       token: process.env.GITHUB_TOKEN,       repo: process.env.ELOG_GITHUB_REPO,       prefixKey: '',       host: 'cdn.jsdelivr.net'     }   },  
然后在Github 流水线中注入ELOG_GITHUB_USER和ELOG_GITHUB_REPO，不需要配置，流水线可自动获取
 image.png (view on web)

**Reply by LetTTGACO** [reply]
看Elog 文件 或者本地试一试就知道了

**Comment by xinkeji** [comment]
本地早就试过了，不行



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年1月16日(周二) 下午5:51
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [LetTTGACO/elog] github图床。变量有问题，github不允许设置github开头的密钥 (Discussion #58)




 
Closed #58 as resolved.

**Reply by CCKNBC** [reply]
祝你成功吧

**Comment by xinkeji** [comment]
我这个是不是直接用cos就直接可以了



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年1月16日(周二) 下午5:54
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [LetTTGACO/elog] github图床。变量有问题，github不允许设置github开头的密钥 (Discussion #58)




 
祝你成功吧

**Reply by LetTTGACO** [reply]
但是问题的关键是你没在流水线文件中设置，并不是在仓库的环境变量中，都拿不到，你用什么都一样的结果

**Comment by LetTTGACO** [comment]
而且 你的配置文件写错了
```
image: {
    enabled: true, // 这里是 enable 不是 enabled，难怪触发不了图片上传
    platform: 'github',
    github: {
      user: process.env.LGH_USER,
      token: process.env.GITHUB_TOKEN,
      repo: process.env.LGH_REPO,
      prefixKey: '',
      host: 'cdn.jsdelivr.net'
    }
  },
```

**Comment by xinkeji** [comment]
好吧，放弃了



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年1月16日(周二) 晚上6:10
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [LetTTGACO/elog] github图床。变量有问题，github不允许设置github开头的密钥 (Discussion #58)




 
而且 你的配置文件写错了
 image: {     enabled: true, // 这里是 enable 不是 enabled，难怪触发不了图片上传     platform: 'github',     github: {       user: process.env.LGH_USER,       token: process.env.GITHUB_TOKEN,       repo: process.env.LGH_REPO,       prefixKey: '',       host: 'cdn.jsdelivr.net'     }   },
